tests/tools/test_mypy_works.py

```diff
@@ -74,3 +74,17 @@ def test_mypy_with_local_import(capsys: "CaptureFixture") -> None:
     out, _ = capsys.readouterr()
     expected = "Success: no issues found in 1 source file\n"
     assert out == expected
+
+
+def test_notebook_doesnt_shadow_python_module(capsys: "CaptureFixture") -> None:
+    """Check that notebook with same name as a Python file doesn't overshadow it."""
+    cwd = os.getcwd()
+    try:
```

Comment:
Instead of `try..finally` how about writing a context manager

```python
from contextlib import contextmanager
@contextmanager
def change_dir(cwd: str):
    os.chdir(os.path.join("tests", "data"))
    yield
    os.chdir(cwd)

with change_dir(cwd):
    with pytest.raises(SystemExit):
                main(["mypy", "t.ipynb"])
```

I find using context manager more relevant compared to `try..finally` semantically.

Reply:
Does this work? Because if the `main` command throws an error, we won't reach `os.chdir(cwd)`, while `finally` is always executed

Reply:
How about making it a fixture with `try..finally` for maximum reusability?

Reply:
will do if this needs doing elsewhere too, else IMO it's simple enough like this